sunpy/map/mapbase.py

```diff
@@ -1524,46 +1524,22 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
         """
         # Check if both context managers are active
         if ACTIVE_CONTEXTS.get('propagate_with_solar_surface', False) and ACTIVE_CONTEXTS.get('assume_spherical_screen', False):
-            warn_user("Using propagate_with_solar_surface and SphericalScreen together result in loss of off-disk data.")
-
-        try:
-            import reproject
-        except ImportError as exc:
-            raise ImportError("This method requires the optional package `reproject`.") from exc
-
+            warn_user("Using propagate_with_solar_surface and assume_spherical_screen together result in loss of off-disk data.")
         if not isinstance(target_wcs, astropy.wcs.WCS):
-            target_wcs = astropy.wcs.WCS(target_wcs)
-
-        # Select the desired reprojection algorithm
-        functions = {'interpolation': reproject.reproject_interp,
-                     'adaptive': reproject.reproject_adaptive,
-                     'exact': reproject.reproject_exact}
-        if algorithm not in functions:
-            raise ValueError(f"The specified algorithm must be one of: {list(functions.keys())}")
-        func = functions[algorithm]
-
-        # reproject does not automatically grab the array shape from the WCS instance
-        if target_wcs.array_shape is not None:
-            reproject_args.setdefault('shape_out', target_wcs.array_shape)
-
-        # Reproject the array
-        output_array = func(self, target_wcs, return_footprint=return_footprint, **reproject_args)
-        if return_footprint:
-            output_array, footprint = output_array
-
-        # Create and return a new GenericMap
-        outmap = GenericMap(output_array, meta=target_wcs.to_header(),
-                            plot_settings=self.plotter.plot_settings)
-
-        # Check rsun mismatch
-        if self.rsun_meters != outmap.rsun_meters:
+            target_wcs = astropy.wcs.WCS(header=target_wcs)
+        if not np.equal(rsun_target:= target_wcs.wcs.aux.rsun_ref, self.rsun_meters.to_value(u.m)):
             warn_user("rsun mismatch detected: "
-                      f"{self.name}.rsun_meters={self.rsun_meters}; {outmap.name}.rsun_meters={outmap.rsun_meters}. "
+                      f"{self.name}.rsun_meters={self.rsun_meters} != {rsun_target} rsun_meters of target WCS."
                       "This might cause unexpected results during reprojection.")
+        # Check if both context managers are active
+        if ACTIVE_CONTEXTS.get('propagate_with_solar_surface', False) and ACTIVE_CONTEXTS.get('assume_spherical_screen', False):
+            warn_user("Using propagate_with_solar_surface and SphericalScreen together result in loss of off-disk data.")
 
-        if return_footprint:
-            return outmap, footprint
-        return outmap
+        reprojected_output = super().reproject_to(target_wcs,
```

Comment:
So this causes some tests to fail:

They are: sunpy/map/tests/test_header_helper.py::test_make_heliographic_header

```
self = <sunpy.map.sources.sdo.AIAMap object at 0x74214f729b50>
SunPy Map
---------
Observatory:             SDO
Instrument:              AIA 3
Detector:                AIA
Measurement:             171.0 Angstrom
Wavelength:              171.0 Angstrom
Observation Date:        2011-02-15 00:00:00
Reference Date:          2011-02-15 00:00:01
Exposure Time:           2.000191 s
Pixel Dimensions:                [128. 128.]
Coordinate System:       helioprojective
Scale:                   [19.183648 19.183648] arcsec / pix
Reference Pixel:         [63.5 63.5] pix
Reference Coord:         [-4.53217221  2.86557481] arcsec                   
array([[-1.25,  0.  ,  1.  , ...,  0.  ,  0.5 , -0.75],
       [ 0.75, -0.25, -0.5 , ...,  0.25,  0.  , -0.25],
       [ 0.  ,  0.5 ,  1.75, ...,  0.  ,  0.5 ,  0.  ],
       ...,
       [ 1.  ,  0.25, -0.25, ...,  0.  ,  0.  ,  0.  ],
       [-0.25,  0.  , -0.5 , ...,  0.75, -0.75,  0.  ],
       [ 0.75,  1.5 , -0.75, ...,  0.  , -0.5 ,  0.5 ]], dtype='>f8')
target_wcs = WCS Keywords

Number of WCS axes: 2
CTYPE : 'HGLN-CAR' 'HGLT-CAR' 
CRVAL : np.float64(0.0) np.float64(0.0) 
CRPIX : np.float64(180.5) np.float64(90.5) 
PC1_1 PC1_2  : np.float64(1.0) np.float64(-0.0) 
PC2_1 PC2_2  : np.float64(0.0) np.float64(1.0) 
CDELT : np.float64(1.0) np.float64(1.0) 
NAXIS : 360  180
algorithm = 'interpolation', shape_out = None, return_footprint = False, reproject_args = {}, reproject_adaptive = <function reproject_adaptive at 0x74214f6fe3e0>
reproject_exact = <function reproject_exact at 0x74214f49f1a0>, reproject_interp = <function reproject_interp at 0x74214f49ed40>, has_celestial = <function has_celestial at 0x74214f6ff420>
algorithms = {'adaptive': <function reproject_adaptive at 0x74214f6fe3e0>, 'exact': <function reproject_exact at 0x74214f49f1a0>, 'interpolation': <function reproject_interp at 0x74214f49ed40>}
low_level_target_wcs = WCS Keywords

Number of WCS axes: 2
CTYPE : 'HGLN-CAR' 'HGLT-CAR' 
CRVAL : np.float64(0.0) np.float64(0.0) 
CRPIX : np.float64(180.5) np.float64(90.5) 
PC1_1 PC1_2  : np.float64(1.0) np.float64(-0.0) 
PC2_1 PC2_2  : np.float64(0.0) np.float64(1.0) 
CDELT : np.float64(1.0) np.float64(1.0) 
NAXIS : 360  180

    def reproject_to(self, target_wcs, algorithm='interpolation', shape_out=None, return_footprint=False, **reproject_args):
        """
        Reprojects the instance to the coordinates described by another WCS object.
    
        Parameters
        ----------
        target_wcs : `astropy.wcs.wcsapi.BaseHighLevelWCS`, `astropy.wcs.wcsapi.BaseLowLevelWCS`,
            or `astropy.io.fits.Header`
            The WCS object to which the `ndcube.NDCube` is to be reprojected.
    
        algorithm: {'interpolation' | 'adaptive' | 'exact'}
            The algorithm to use for reprojecting.
            When set to "interpolation" `~reproject.reproject_interp` is used,
            when set to "adaptive" `~reproject.reproject_adaptive` is used and
            when set to "exact" `~reproject.reproject_exact` is used.
    
        shape_out: `tuple`, optional
            The shape of the output data array. The ordering of the dimensions must follow NumPy
            ordering and not the WCS pixel shape.
            If not specified, `~astropy.wcs.wcsapi.BaseLowLevelWCS.array_shape` attribute
            (if available) from the low level API of the ``target_wcs`` is used.
    
        return_footprint : `bool`
            If `True` the footprint is returned in addition to the new `~ndcube.NDCube`.
            Defaults to `False`.
    
        **reproject_args
            All other arguments are passed through to the reproject function
            being called. The function being called depends on the
            ``algorithm=`` keyword argument, see that for more details.
    
        Returns
        -------
        reprojected_cube : `ndcube.NDCube`
            A new resultant NDCube object, the supplied ``target_wcs`` will be the ``.wcs`` attribute of the output `~ndcube.NDCube`.
    
        footprint: `numpy.ndarray`
            Footprint of the input array in the output array.
            Values of 0 indicate no coverage or valid values in the input
            image, while values of 1 indicate valid values.
    
        See Also
        --------
    
        reproject.reproject_interp
        reproject.reproject_adaptive
        reproject.reproject_exact
    
        Notes
        -----
        This method doesn't support handling of the ``mask``, ``extra_coords``, and ``uncertainty`` attributes yet.
        However, ``meta`` and ``global_coords`` are copied to the output `ndcube.NDCube`.
        """
        try:
            from reproject import reproject_adaptive, reproject_exact, reproject_interp
            from reproject.wcs_utils import has_celestial
        except ModuleNotFoundError:
            raise ImportError(f"The {type(self).__name__}.reproject_to method requires the `reproject` library to be installed.")
    
        algorithms = {
            "interpolation": reproject_interp,
            "adaptive": reproject_adaptive,
            "exact": reproject_exact,
        }
    
        if algorithm not in algorithms.keys():
            raise ValueError(f"{algorithm=} is not valid, it must be one of {', '.join(algorithms.keys())}.")
    
        if isinstance(target_wcs, Mapping):
            target_wcs = WCS(header=target_wcs)
    
        low_level_target_wcs = utils.wcs.get_low_level_wcs(target_wcs, 'target_wcs')
    
        # 'adaptive' and 'exact' algorithms work only on 2D celestial WCS.
        if algorithm == 'adaptive' or algorithm == 'exact':
            if low_level_target_wcs.pixel_n_dim != 2 or low_level_target_wcs.world_n_dim != 2:
                raise ValueError('For adaptive and exact algorithms, target_wcs must be 2D.')
    
            if not has_celestial(target_wcs):
                raise ValueError('For adaptive and exact algorithms, '
                                 'target_wcs must contain celestial axes only.')
    
        if not utils.wcs.compare_wcs_physical_types(self.wcs, target_wcs):
>           raise ValueError('Given target_wcs is not compatible with this NDCube, the physical types do not match.')
E           ValueError: Given target_wcs is not compatible with this NDCube, the physical types do not match.
```
This I think has to be patched in ndcube?